scripts/markdown/check-edit-links.js

```diff
@@ -0,0 +1,163 @@
+const fs = require('fs').promises;
+const path = require('path');
+const fetch = require('node-fetch-2');
+const editUrls = require('../../config/edit-page-config.json');
+const { pause } = require('../dashboard/build-dashboard');
+
+const ignoreFiles = [
+  'reference/specification/v2.x.md',
+  'reference/specification/v3.0.0-explorer.md',
+  'reference/specification/v3.0.0.md'
+];
+
+/**
+ * Process a batch of URLs to check for 404s
+ * @param {object[]} batch - Array of path objects to check
+ * @returns {Promise<string[]>} Array of URLs that returned 404
+ */
+async function processBatch(batch) {
+  const TIMEOUT_MS = 5000;
+  return Promise.all(
+    batch.map(async ({ filePath, urlPath, editLink }) => {
+      try {
+        if (!editLink || ignoreFiles.some((ignorePath) => filePath.endsWith(ignorePath))) return null;
+
+        const controller = new AbortController();
+        /* istanbul ignore next */
+        const timeout = setTimeout(() => controller.abort(), TIMEOUT_MS);
+        const response = await fetch(editLink, {
+          method: 'HEAD',
+          signal: controller.signal
+        });
+        clearTimeout(timeout);
+        if (response.status === 404) {
+          return { filePath, urlPath, editLink };
+        }
+        return null;
+      } catch (error) {
+        return Promise.reject(new Error(`Error checking ${editLink}: ${error.message}`));
+      }
+    })
+  );
+}
+
+/**
+ * Check all URLs in batches
+ * @param {object[]} paths - Array of all path objects to check
+ * @returns {Promise<string[]>} Array of URLs that returned 404
+ */
+async function checkUrls(paths) {
+  const result = [];
+  const batchSize = 5;
```

Comment:
Kindly make it configurable using env variable.

Reply:
Done